src/Compilers/VisualBasic/Portable/Language/LanguageFeature.vb

```diff
@@ -0,0 +1,296 @@
+        LineContinuationComments
+        TypeOfIsNot
+        YearFirstDateLiterals
+        WarningDirectives
+        PartialModules
+        PartialInterfaces
+        ImplementingReadonlyOrWriteonlyPropertyWithReadwrite
+        DigitSeparators
+        BinaryLiterals
+        Tuples
+        IOperation
+        InferredTupleNames
+    End Enum
+
+    <HideModuleName>
+    Friend Module FeatureExtensions
+
+        <Extension>
+        Private Function RequiresExplicitFeatureFlag(feature As Feature) As Boolean
+            Return feature = Feature.IOperation
+        End Function
+
+        <Extension>
+        Friend Function GetFeatureFlag(feature As Feature) As String
+            Select Case feature
+                Case Feature.IOperation
+                    Return "IOperation"
+
+                Case Else
+                    Return Nothing
+            End Select
+        End Function
+
+        <Extension>
+        Friend Function GetLanguageVersion(feature As Feature) As LanguageVersion
+
+            Select Case feature
+                Case Feature.AutoProperties,
+                     Feature.LineContinuation,
+                     Feature.StatementLambdas,
+                     Feature.CoContraVariance,
+                     Feature.CollectionInitializers,
+                     Feature.SubLambdas,
+                     Feature.ArrayLiterals
+                    Return LanguageVersion.VisualBasic10
+
+                Case Feature.AsyncExpressions,
+                     Feature.Iterators,
+                     Feature.GlobalNamespace
+                    Return LanguageVersion.VisualBasic11
+
+                Case Feature.NullPropagatingOperator,
+                     Feature.NameOfExpressions,
+                     Feature.InterpolatedStrings,
+                     Feature.ReadonlyAutoProperties,
+                     Feature.RegionsEverywhere,
+                     Feature.MultilineStringLiterals,
+                     Feature.CObjInAttributeArguments,
+                     Feature.LineContinuationComments,
+                     Feature.TypeOfIsNot,
+                     Feature.YearFirstDateLiterals,
+                     Feature.WarningDirectives,
+                     Feature.PartialModules,
+                     Feature.PartialInterfaces,
+                     Feature.ImplementingReadonlyOrWriteonlyPropertyWithReadwrite
+                    Return LanguageVersion.VisualBasic14
+
+                Case Feature.Tuples,
+                    Feature.BinaryLiterals,
+                    Feature.DigitSeparators
+                    Return LanguageVersion.VisualBasic15
+
+                Case Feature.InferredTupleNames
+                    Return LanguageVersion.VisualBasic15_3
+                Case Else
+                    ' Return a "dummy" version for any feature that requires an explicit feature flag.
+                    If feature.RequiresExplicitFeatureFlag Then Return CType(_RequireExplictFeatureFlag, LanguageVersion)
```

Comment:
>CType [](start = 71, length = 5)

We should not be producing a value of type LanguageVersion that is not a member of the enum. If we were to require an API to tell us if a feature "requires" a feature flag, we would create a new API and call it "RequiresFeatureFlag".